src/features/favorites/pages/reducer.ts

```diff
@@ -1,26 +1,34 @@
-import { FavoriteResponse } from 'api/gen'
+import { FILTER_OPTIONS } from 'features/favorites/pages/Filters'
 
 export type FavoritesParameters = {
-  favorites: Array<FavoriteResponse>
+  filter: keyof typeof FILTER_OPTIONS | undefined
 }
 
 export type FavoritesState = FavoritesParameters & {
   showResults: boolean
+  filter: keyof typeof FILTER_OPTIONS | undefined
   dispatch: (action: Action) => void
 }
 
 export const initialFavoritesState: FavoritesState = {
   showResults: false,
-  favorites: [],
+  filter: undefined,
   dispatch: () => null,
 }
 
-export type Action = { type: 'SHOW_RESULTS'; payload: boolean }
+export type Action =
+  | { type: 'INIT' }
+  | { type: 'SHOW_RESULTS'; payload: boolean }
+  | { type: 'SET_FILTER'; payload: keyof typeof FILTER_OPTIONS | undefined }
 
 export const favoritesReducer = (state: FavoritesState, action: Action): FavoritesState => {
   switch (action.type) {
+    case 'INIT':
+      return { ...initialFavoritesState, dispatch: state.dispatch, showResults: state.showResults }
```

Comment:
Ca ne devrait pas être comme cela ?
```suggestion
      return initialFavoritesState
```

Reply:
Le dispatch du `initialFavoriteState` n'en est pas un, c'est depuis qu'on sorte le dispatch dans le state.

Reply:
ok